spec/lib/sendgrid_helper_spec.rb

```diff
@@ -2,62 +2,116 @@
 require 'sendgrid_helper'
 
 RSpec.describe SendgridHelper do
-  context "spam reports" do
-    context "error handling" do
-      [Curl::Err::CurlError, JSON::ParserError].each do |exception_class|
-        it "marks the email as valid if there's an error (#{exception_class})" do
-          expect(Curl::Easy).to receive(:perform).and_raise(exception_class)
+  context 'with sendgrid configured' do
+    around do |example|
+      smtp_settings = Rails.configuration.action_mailer.smtp_settings
+      Rails.configuration.action_mailer.smtp_settings = {
+        user_name: 'test_smtp_username',
+        password: 'test_smtp_password'
+      }
+      example.run
+      Rails.configuration.action_mailer.smtp_settings = smtp_settings
+    end
+
+    context "spam reports" do
+      context "error handling" do
+        [Curl::Err::CurlError, JSON::ParserError].each do |exception_class|
+          it "marks the email as valid if there's an error (#{exception_class})" do
+            allow(Curl::Easy).to receive(:perform).and_raise(exception_class)
+            expect(SendgridHelper.spam_reported?('test@example.com')).to be_falsey
```

Comment:
Two things:

##### 1.
Does `spam_reported?` always return instances of the `TrueClass` or `FalseClass`?

If so its better to show this in the assertion by using `be true` or `be false`, it reveals with less ambiguity the return value of the method. This applies to any other predicate methods that return only `true` or `false` values.

##### 2.
I'd suggest using RSpec's `described_class` when writing expectations about the current 'thing' in question. Its intended to return the class or module that is being described. If you aren't happy with the fact that it uses class in its name when talking about a `Module` then perhaps set it as the `subject` - but keep it consistent with the way other modules have been tested.

@threedaymonk @SteveMarshall do you guys agree?